netcore/System.Private.CoreLib/src/System.Runtime.Loader/AssemblyLoadContext.cs

```diff
@@ -134,5 +134,53 @@ internal static Assembly DoAssemblyResolve (string name)
 		{
 			return ResolveSatelliteAssembly (gchALC, new AssemblyName (assemblyName));
 		}
+
+#region Copied from AssemblyLoadContext.CoreCLR.cs
+		// This method is called by the VM. TODO: wire up to Mono
+		private static RuntimeAssembly? OnResourceResolve(RuntimeAssembly assembly, string resourceName)
+		{
+			return InvokeResolveEvent(ResourceResolve, assembly, resourceName);
+		}
+
+		// This method is called by the VM. TODO: wire up to Mono
+		private static RuntimeAssembly? OnTypeResolve(RuntimeAssembly assembly, string typeName)
+		{
+			return InvokeResolveEvent(TypeResolve, assembly, typeName);
+		}
+
+		// This method is called by the VM.
+		private static RuntimeAssembly? OnAssemblyResolve(RuntimeAssembly assembly, string assemblyFullName)
+		{
+			return InvokeResolveEvent(AssemblyResolve, assembly, assemblyFullName);
+		}
+
+		private static RuntimeAssembly? InvokeResolveEvent(ResolveEventHandler? eventHandler, RuntimeAssembly assembly, string name)
+		{
+			if (eventHandler == null)
+				return null;
+
+			var args = new ResolveEventArgs(name, assembly);
+
+			foreach (ResolveEventHandler handler in eventHandler.GetInvocationList())
+			{
+				Assembly? asm = handler(AppDomain.CurrentDomain, args);
+				RuntimeAssembly? ret = GetRuntimeAssembly(asm);
+				if (ret != null)
+					return ret;
+			}
+
+			return null;
+		}
+#endregion
+#region Copied from AssemblyLoadContext.CoreCLR.cs, modified until our AssemblyBuilder implementation is functional
+		private static RuntimeAssembly? GetRuntimeAssembly(Assembly? asm)
+		{
+			return
+				asm == null ? null :
+				asm is RuntimeAssembly rtAssembly ? rtAssembly :
+				//asm is System.Reflection.Emit.AssemblyBuilder ab ? ab.InternalAssembly :
```

Comment:
Is the only issue that we don't have an `InternalAssembly` property? could we add one that throws NIE and a FIXME to note to come back to it?

Reply:
I can do that if we're sure we want to use an InternalAssemblyBuilder the same way CoreCLR has. See: https://github.com/dotnet/coreclr/blob/52651008a5fb72eb678467cd2eb42aac4e5334e8/src/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs#L33

Reply:
I see.  Well then I guess we'll have to come back here when we work on SRE for netcore.

Reply:
Yeah, looking at it again I think the best option here is to not move this function to shared yet so we don't play ping-pong with the code, then re-evaluate at a later date. I'll update the comment to be clearer though.